python/common/org/Python.java

```diff
@@ -814,15 +814,9 @@ public static void initializeModule(java.lang.Class cls, java.util.Map<java.lang
         } else if (class_or_type_or_tuple == null) {
             throw new org.python.exceptions.TypeError("isinstance expected 2 arguments, got 1");
         } else if (class_or_type_or_tuple instanceof org.python.types.Tuple) {
-            java.util.List<org.python.Object> classes = ((org.python.types.Tuple) class_or_type_or_tuple).value;
-            for (org.python.Object klass: classes) {
-                if (((org.python.types.Bool) org.Python.isinstance(object, klass).__bool__()).value) {
-                    return new org.python.types.Bool(true);
-                };
-            }
-            return new org.python.types.Bool(false);
+            return org.Python.issubclass(object.type(), class_or_type_or_tuple);
```

Comment:
In Python 2/3, you can override the `isinstance` check using a meta class. So this code will have to be changed at a later date, but that can be done for a separate issue.